### DISCUSSION TITLE
QUESTION: How can I martinize very big protein structures accurately? I've tried using martinize with structures with atoms >99999 and I received errors because it was unable to read the pdb file.

### DISCUSSION THREAD

**Discussion by Boupalik** [marrink-lab/vermouth-martinize#606] [General]
What is the best way to martinize very big structures with more atoms than 99999. I've been trying to do this by splitting my pdb file into various chains and martinizing and combining them, only to face linc errors and infinite energy. What ways do you guys use to martinize very big structures and face no errors in the gromacs simulation?

**Comment by pckroon** [comment]
How sure are you your coordinates are sane? If gromacs runs (i.e. grompp doesn't throw errors/warnings), and you still get infinite energies and LINCS warnings I guess you have clashes.

**Reply by amiel-ab** [reply]
Hi @pckroon @Boupalik 
I got the exact same issue I believe it might be due to various component structured with elastic network "bumping" to each other (in contrast with one protein with elastic)
Any ideas on that ?
I was able to go through minimisation (hardly by changing the lincs .mdp parameters ) but got stuck in equilibration.
My structure is coming from atomistic simulation that has been equilibrated.

**Reply by pckroon** [reply]
It's been forever since I've had to run a simulation, but the fact that it doesn't work after you start from an equilibrated atomistic structure has me a bit concerned. Maybe @csbrasnett has any insights here?

**Reply by csbrasnett** [reply]
If you're able to energy minimise the system but not start dynamics, it could be worth doing timestep incremental equilibrations, doing a series of eg. 0.002, 0.005, 0.01 ps timesteps starting from each previous step (remember continuation  = no and gen_vel = -1 in your mdps after the first run!).

Otherwise, it might be worth checking your elastic network by varying the force constant or cutoff, or using a different -eunit when martinizing them.

btw, with the new version of [cgmartini.nl](https://cgmartini.nl/) we have a new [martini discussion forum](https://github.com/orgs/Martini-Force-Field-Initiative/discussions) which might be another useful place to discuss such challenges with martini simulations

**Reply by amiel-ab** [reply]
I'll follow those advices.
Thanks a lot @pckroon and @csbrasnett !

**Comment by fpontes22** [comment]
Hi!
I think, in my facing this same type of problem. Independent if I'm using an minimized structure in vacuum or solution (both minimized with Fmax around 1e+02), when I run martinize2, I had problems with the chain C. In this chain the atom numbering goes to zero again after the structure reached 10000 atoms. I checked the pdb files created by dssp   during the process, and the pdb file referent to this chain has a starter residue with number 6205 and atom number 0 with an incomplete length. 
The resolution transformation finishes with the message: "ValueError: The sequence length does not match the number of residues. "

Thank you in advance for any help.

Fred

**Reply by pckroon** [reply]
Nowadays martinize2 can also read .cif files, which are not fixed width and should have no issues with large atom numbers. Maybe worth a try?

**Reply by fpontes22** [reply]
Thank you for the suggestion. This is very useful for my future simulations. I solved my problem writing a small code in Python that restart the residues number every time that a new chain starts. I could create topology and structure at coarse-grain level without problems.

Thank you all for the amazing support.

**Reply by pckroon** [reply]
Hmmn. It might be a good addition to do that internally, always. On https://github.com/marrink-lab/vermouth-martinize/blob/master/vermouth/dssp/dssp.py#L301 the dssp pipeline writes the system to the temporary pdb file that dssp then reads. It may be a decent option to change all resids to residue indices (1-...) there. Would you be willing to implement that and open a PR along those lines?

**Reply by fpontes22** [reply]
Hi Peter!
It would be amazing. I'm quite busy this week, but next Monday I can start this. Thank you for the invitation.

**Reply by pckroon** [reply]
No rush whatsoever <3
